zkvm/src/signature/multikey.rs

```diff
@@ -0,0 +1,58 @@
+use crate::errors::VMError;
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+
+#[derive(Clone)]
+pub struct Multikey {
+    transcript: Transcript,
+    X_agg: VerificationKey,
+}
+
+impl Multikey {
+    pub fn new(pubkeys: Vec<VerificationKey>) -> Result<Self, VMError> {
+        // Create transcript for Multikey
+        let mut transcript = Transcript::new(b"ZkVM.aggregated-key");
+
+        // Hash in pubkeys
+        // L = H(X_1 || X_2 || ... || X_n)
```

Comment:
for disambiguation we need a length prefix before committing any variable-length lists: `commit_u64("n", pubkeys.len())`